frontends/rpc/rpc_frontend.cc

```diff
@@ -209,7 +209,7 @@ struct RpcModule : RTLIL::Module {
 					if (name_mangling.count(cell->type.str()))
 						cell->type = name_mangling[cell->type.str()];
 
-			for (auto module : derived_design->modules_) {
+			for (const auto &module : derived_design->modules_) {
```

Comment:
This creates a reference to a pair containing an IdString (just and int) and a pointer. The code in this block is only using module.first (the int) and module.second (the pointer) directly, never the whole pair. In what way is this code improved by creating a reference to that pair instead of loading this two members right here directly?

Reply:
Similar here, the code output in this case is the same https://godbolt.org/z/FQeBge

Reply:
Same as before. `module.first` is being used, then external functions are called that the compiler has no visibility into, then `module.first` is used again, forcing the compiler to re-load the values from memory.

Your compiler explorer example is carefully avoiding doing exactly that. Thus it has no relevance whatsoever for explaining what will happen with this code.

Reply:
Looking at the actual object file we also see more activity with the code that copies vs. the const reference due to the reference counting business:

(compare this vs. changing line 212 with `for (const auto &module : ...`)
```
make && objdump -l -C -d -S ./frontends/rpc/rpc_frontend.o | awk '/rpc_frontend.cc:212/ { do_print=1; }  { if (do_print) print $0; } /rpc_frontend.cc:226/ { do_print=0; }' > rpcfrontend.1
```